ed_gui_server/src/gui_server_plugin.cpp

```diff
@@ -537,4 +562,146 @@ bool GUIServerPlugin::srvInteract(const ed_gui_server_msgs::Interact::Request& r
+            }
+        }
+    }
+
+    // Small padding around the entities in view
+    p_max.x *= (p_max.x>=0) ? 1.01 : 1/1.01;
+    p_max.y *= (p_max.y>=0) ? 1.01 : 1/1.01;
+    p_min.x *= (p_min.x>=0) ? 1/1.01 : 1.01;
+    p_min.y *= (p_min.y>=0) ? 1/1.01 : 1.01;
+
+    geo::Vec2 range = p_max - p_min;
+    geo::Vec2 center = 0.5*(p_max + p_min);
+    double dist = 1000; // Value doesn't influence the generated img
+
+    geo::Pose3D cam_pose;
+    cam_pose.t = center.projectTo3d();
+    cam_pose.t.z = dist;
+    cam_pose.R = geo::Matrix3::identity();
+    if (range.y > range.x)
+    {
+        cam_pose.R.setRPY(0, 0, -M_PI_2);
+        std::swap(range.x, range.y);
+    }
+
+    uint width = req.image_width ? req.image_width : req.DEFAULT_WIDTH;
+    uint height = req.image_height ? req.image_height : req.DEFAULT_HEIGHT;
+    double focal_length = std::min(width/range.x, height/range.y); // Pixels per meter
+
+    geo::DepthCamera cam;
+    cam.setFocalLengths(focal_length * dist, focal_length * dist);
+    cam.setOpticalCenter(width / 2 + 0.5, height / 2 + 0.5);
+    cam.setOpticalTranslation(0, 0);
+
+    cv::Mat depth_image = cv::Mat(height, width, CV_32FC1, 0.0);
+    cv::Scalar background_color;
+    if (req.background == req.WHITE)
+        background_color = cv::Scalar(255, 255, 255);
+    else if (req.background == req.BLACK)
+        background_color = cv::Scalar(20, 20, 20); // Not completely black
+    else
+        // Default black
+        background_color = cv::Scalar(20, 20, 20); // Not completely black
+    cv::Mat image = cv::Mat(height, width, CV_8UC3, background_color);
+
+    const geo::Pose3D& cam_pose_inv = cam_pose.inverse();
+    ed::renderWorldModel(*world_model_, ed::ShowVolumes::NoVolumes, cam, cam_pose_inv, depth_image, image, true);
+
+    if (req.print_labels)
+    {
+        const cv::Scalar text_color = cv::Scalar(255, 255, 255) - background_color;
+        for (ed::WorldModel::const_iterator it = world_model_->begin(); it != world_model_->end(); ++it)
+        {
+            const ed::EntityConstPtr& e = *it;
+            const std::string& id = e->id().str();
+
+            if (e->shape() && e->has_pose() && !e->hasFlag("self") && (id.size() < 5 || id.substr(id.size() - 5) != "floor")) // Filter ground plane
+            {
+                geo::Vector3 center = e->pose().getOrigin();
+                center.z = 0;
+                cv::Point center_cv = cam.project3Dto2D(cam_pose_inv * center);
+                int tmp_baseline;
+                cv::Size textSize = cv::getTextSize(id, cv::FONT_HERSHEY_COMPLEX_SMALL, 1, 1, &tmp_baseline);
+                center_cv.x -= textSize.width/2;
+                center_cv.y += textSize.height/2;
+                cv::putText(image, id, center_cv, cv::FONT_HERSHEY_COMPLEX_SMALL, 1, text_color, 1);
+            }
+        }
+    }
+
+    rgbd::convert(image, res.map);
+    res.pixels_per_meter_width = res.pixels_per_meter_height = focal_length;
```

Comment:
```suggestion
    res.pixels_per_meter_width = focal_length;
    res.pixels_per_meter_height = focal_length;
```
Performance makes no difference with modern compilers. So I would prefer the easier to read version.